Hand of God : A spin-off of cult [DON'T MERGE/93% Finished]

Neerti opened the pull request:
Main idea can be seen here. http://www.ss13.eu/phpbb/viewtopic.php?f=42&t=4184

This was polled and passed with flying colors on oldforums.

Thanks to Pete, Sawu, wb, RemieRichards, Iamgoofball for helping out with the code.
Thanks to Neinhaus, MrSnapwalk, Wjohnston for sprites.
Thanks to the people in #coderbus and #nanobus for answering my dumb code questions.

If I forgot someone let me know.

Remaining things to do:
- [ ] Objectives
  - [x] Follower Block objective
  - [ ] Deicide objective
  - [ ] Build objective
  - [x] Follower Escape objective
  - [ ] Sacrifice Prophet objective
- [ ] Bugfixes 
  - [ ] Fix god say code
  - [ ] Fix pylons shooting at gods
  - [ ] Fix admin player panel not showing deities
  - [ ] Fix structure building proc not stopping if a structure already exists
  - [ ] Fix rangefinding for abiltiies
  - [ ] Fix god icon
  - [ ] Fix follower counting
- [ ] Playtesting
  - [ ] Fix more bugs
- [ ] Merge conflict resolution

CURRENT BUGS: Carn yells at me from byond the code because I use transfer_to() a mob/camera, but the mind needs to transfer.  Blob somehow gets around this.  Not sure if it's worth it to make a snowflake transfer or to just remove the error as it doesn't appear to actually cause any damage.
The proc to build a structure doesn't stop if a structure already exists in it's location.  idk why atm.
Defensepylons try to shoot at gods, this is bad for obvious reasons.
Talking to other gods and prophets is broken.
Rangefinding for abilities doesn't work.
The icon for gods is visible to all humans.  This isn't intended.
Follower Counting isn't functioning.

<B>Remie Edit: Approved for use of Topic Procs</B>

Neerti committed:
Committing this stuff incase I mess up later.

Neerti committed:
Adds most of the structures, their health system, ability to be damaged.  Most buildings do nothing at this moment.

Atskadan commented:
neato

GunHog commented:
Do you think /tg/ would accept this as well?

Neerti commented:
Since my goal is to not touch cult assets as much as possible it might make it easier to port.

Neerti committed:
Adds more functioning to structures.
Adds ability to tie a god to a nexus.
Adds basic structure building (later, cultists will need to finish it with materials).
Adds wip sprites.

Neerti committed:
Adds dependence for nexus, after placed.  If destroyed, the god dies.
Adds almost complete hud for deity.
Add more stuff for structures and powers.

PinkyGatchabee commented:
excited for this

Neerti committed:
Add point regeneration.

Iamgoofball committed:
updates

Neerti committed:
Merge pull request #1 from Iamgoofball/handofgod

updates

Neerti committed:
Adds nice little icons for the cultists to tell who else is a cultist, at least after that gets coded.

Neerti committed:
Adds default names defined in deity.txt in config/names.  You can still change your name when the game starts, but if you don't choose any you're not stuck with 'deity'.

Iamgoofball committed:
fixes

Neerti committed:
Merge pull request #2 from Iamgoofball/handofgod

updates

Iamgoofball committed:
more shit

Iamgoofball committed:
forgot something

Neerti committed:
Merge pull request #3 from Iamgoofball/handofgod

more shit

Neerti committed:
Adds about 20% of what's needed to make the gamemode work.  Warranty void if used.

Neerti committed:
Merge branch 'cult2' of https://github.com/Neerti/NTstation13 into cult2

Neerti committed:
Adds more to the infrastructure of the gamemode, including adding the red team to the admin antag panel.
Also adds follower icons for red cult.  Blue clue can be copypasta'd later.
Adds foundations of objectives.  Remember only the god gets the objective.

Neerti commented:
Red team is mostly done in terms of infrastructure.  Still need a more robust convert and deconvert proc.

Neerti committed:
Adds deconversion proc.  Adds deconverting to implants and holy water, taken from malkcode.

Neerti commented:
de/convert procs done.  Blue team needs love and buildings and stuff need to be completed.

I'd say I'm 65% done.

Atskadan commented:
65% for 5 days of work is really fucking good.
Make sure you get other people to review it pretty thoroughly though; new gamemodes are a pretty big addition that absolutely needs to be bug-free.

Neerti committed:
Adds support for blue team de/converting and icons.  Rejoice blue team!

Neerti committed:
Finished second team, admin panel can set people to red or blue prophets or followers.  Red team does not see blue team and vise versa.

Neerti committed:
Fixed god talking problems with both say and the talk to gods verb, big thanks to Petethegoat.
Removed some unneeded commented lines.

Neerti commented:
Pete fixed the problem with gods talking.

Neerti committed:
Adds experimental roundstart assigning.  I can't test by myself but I will try to make dreamdaemon stop being up it's ass later and see if it assigns evenly like it should.

Neerti committed:
Makes the convert altAr work properly.  Credit to Sawu for helping out a ton.

Fixes typos with altAr saying altEr and causing errors.

Neerti commented:
Make sure to get Sawu and Goofball in the changelog credits for helping me a lot with bugs.

Neerti committed:
hbl() aka disaster() works now, with ten random events, all bad, that can happen.  It is an endgame ability.

Also minor icon stuff.

Neerti committed:
Makes the deity ability bless work by adding a special snowflake reagent.
Makes sacrificing work.  Certain people yield better results!
Adds lesser and greater gems, which will be used for the forge later.

Neerti commented:
Most structures and powers are finished.

Neerti committed:
Adds construction mechanics for all follower structures.  They require someone to use metal or glass sheets on the unfinished structure, that a deity designates, to finish the structure and having it work.
More icon stuff.
Adds a forgeobj file that'll be used later.

Neerti commented:
Defense pylons are almost finished, but currently suffer from attempting to shoot at a target they can't see, but is within range, as well as a few other minor bugs.

Neerti committed:
Adds working defense pylon, made 95% by wb.

Neerti committed:
Adds crafting to the forge, powered by based Razharas.  Makes crafting better in general with datums and stuff (I need it like this or else assistants will make swords on tables and we'll have the grey crusades)
Adds more fun god names to the deity.txt
I did something else but forgot so yay!

GunHog commented on code/game/gamemodes/handofgod/powers.dm:
```diff
@@ -0,0 +1,346 @@
+	set desc = "Create the foundation of a divine object."
+
+	if(powerc(75,1))
+		var/choice = input("Choose what you wish to create.","Divine structure") as null|anything in list("ward","conduit","forge","convert alter",
```
altar

GunHog commented on code/game/gamemodes/handofgod/structures.dm:
```diff
@@ -0,0 +1,568 @@
+			user << "<span class='notice'>You can't seem to do anything useful with this.</span>"
+			return
+		if(!H || !H.mind) //If nothing is on top of the alter, or the mob has no mind, stop the proc.
+			user << "<span class='danger'>Nobody sentient is on top of the alter, if anything at all.</span>"
```
altar

Iamgoofball commented:
crusader greys II

Atskadan commented:
crusader captains *

Neerti commented:
Crusader Comdoms*

Iamgoofball commented:
The Old Chaplains, Legacy of HoS

Neerti committed:
Worked on forge, with lots of stuff to create, some of which actually work!

Neerti commented:
Added about a third of the forge objects.

vista- commented:
What's the current status on this?

Neerti commented:
Making new forge objects (as in, not just childs of existing things) takes time, and I'd like to have a diverse range of things you can build.

Otherwise when forge is done I just got a few structures and powers to mop up, do roundstart antag assignment, objectives and I should be finished.

GunHog commented:
Any chance of adding the mini-blob event from this: https://github.com/tgstation/-tg-station/pull/3582 to the list of possible events a high powered god can proc?  It still feels that the selection of events is too weak to be worthy of a god.

Also, is the God causing the event told what exactly it did?

Neerti commented:
The god does not know which event it caused. Admins know, however.

Neerti committed:
Added descriptions based on who's examining it.

Neerti committed:
Adds two new structures, which don't work.
Adds new traps, with four types, which DO work.

Still need to make them build-able.

Neerti committed:
Adds more structures.  Also makes smite more snowflake, adds trap sprites.

Neerti committed:
Adds verb to make wards/traps.
Fixes bug with adding materials to build.

Neerti committed:
Comments out this part since the actual item is commented out, so you can't make an item that doesn't exist.  I know not everything's on the list, I'll get to it.

Neerti committed:
Adds roundstart antag selection.  The number of people needed is set to two for debug purposes.
Prophets hearing their god works properly now.
One follower at roundstart turns into a deity now.

Neerti commented:
I'd say I'm 85 to 90% done now.

What remains: Objectives, a few major bugfixes involving defense pylons shooting at deities, range checks for some abilities, finishing abilities, sprites, and some tiny bug fixes.

Neerti committed:
Makes objectives be given to deities at roundstart.
Objectives don't make sense atm but they don't runtime at least!
Fixes serious bug making blue undeconvertable.
Makes prophets unable to be deconverted w/ holywater.

Atskadan commented on code/game/gamemodes/handofgod/forgeobjs.dm:
```diff
@@ -0,0 +1,365 @@
+		usr << "A horrible banner representing disorder and chaos."
+	else
+		usr << "A banner with a toolbox logo, likely assistant-related."
+*/
```
why comment out ;_;

Neerti replied:
It no compile.

And I cba how to make it check for role right now.

Atskadan replied:
if(istype(M, /mob/living/carbon/human) && M.job in list("Assistant")) ?
or if(user.mind && (user.mind.assigned_role == "Assistant"))

Atskadan commented on code/game/gamemodes/handofgod/forgeobjs.dm:
```diff
@@ -0,0 +1,365 @@
+/obj/item/clothing/head/helmet/blessed
+	name = "blessed helmet"
+	desc = "A helmet that seems warm for some reason.  Not warm enough to protect you in space, but it's still nice.  Might be better too?"
+	armor = list(melee = 60, bullet = 25, laser = 60,energy = 20, bomb = 35, bio = 0, rad = 0)
```
still an error here 

> 60,energy

RemieRichards replied:
Uh @Atskadan , that's not an error, byond will see the "," and immediately end what it was doing and move on, if it was an error, it would have been flagged up while Neerti compiled things.

Atskadan replied:
but no other armor values lack a space

RemieRichards replied:
It's not an error, it compiles, there's no point arguing over a space that doesn't change how the code compiles.

Atskadan replied:
ill put a space in YOUR BRAIN :gun:

RemieRichards replied:
no

Atskadan replied:
try and stop me

ZomgPonies commented:
Once this is ready for merge, please for the love of christ compress it down to a single commit.

ZomgPonies commented:
As Atskadan removed his post saying "or, how about you don't barge into other people's codebases and try to tell them what to do to make it easier for you to steal it?" I'll answer on that since everyone got his rant by email anyhow.

First, get the stick out of your ass. If you don't want the code public, don't make the codebase public. Secondly from what I understand, this codebase was created due to dissent in the way /tg/ was being maintained. That means it was designed as much for downstream servers as for NTStation, hence the fact that 36 commits MIGHT be worth compressing, -especially- if it's modular and you intend to try to get it merged into /tg/.

Lastly as for me 'stealing' your public code, why don't you ask goofball who came into IRC not 6 hours ago to ask me if he could grab my new nations gamemode? Learn to play ball with the community.

tl;dr, go fuck yourself Atskadan.

Atskadan commented:
I'm not going to get into an argument with you about how I disapprove of you or your codebase in this PR, and that's why I deleted my post. It's not appropriate to derail this PR over petty personal dislike, so I'm going to ask you to stop right now. If you want to argue, bring it somewhere else.

Iamgoofball commented:
knock it off @dannno 

you're not helping at all

Neerti commented:
Objectives are really confusing so this might take a bit.

Neerti commented:
I'm still working on this, just fyi.

Iamgoofball commented:
merge conflicts

i'm thinking a PR broke everyone's shit recently

RemieRichards commented:
@Iamgoofball 

> [DON'T MERGE/90% Finished]

Iamgoofball commented:
@RemieRichards

yeah but a fuckton of PRs suddenly got merge conflicts.

RemieRichards commented:
@Iamgoofball No doubt my mutations overhaul, 79 files changed.

Iamgoofball commented:
> 79 files changed

r.i.p

Neerti committed:
Commiting half of objective code so I can pick up later without risk.
Also fixes a few minor bugs.

Neerti commented:
I'm abandoning this project, the NTStation Project, and ss13, so this will never be finished.  Anyone's free to use the code to finish it or for whatever.
I'm sorry for disappointing everyone.

vista- commented:
404 not found
On 26 Jul 2014 17:30, "Neerti" notifications@github.com wrote:

> Closed #404 https://github.com/NTStation/NTstation13/pull/404.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NTStation/NTstation13/pull/404#event-145924003.

GunHog commented:
NEERTI WHY WHY WHY :-1:x :100:

Iamgoofball commented:
RIP ARTYOM RIP ARTYOM CODE DELTA CODE DELTA

optimumtact commented:
Happens to the best :tired_face:
